_inc/lib/core-api/class.jetpack-core-api-widgets-endpoints.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Widget information getter endpoint.
+ *
+ */
+class Jetpack_Core_API_Widget_Endpoint {
+
+	/**
+	 * @since 5.5.0
+	 *
+	 * @param WP_REST_Request $request {
+	 *     Array of parameters received by request.
+	 *
+	 *     @type string $id Widget id.
+	 * }
+	 *
+	 * @return WP_REST_Response|WP_Error A REST response if the request was served successfully, otherwise an error.
+	 */
+	public function process( $request ) {
+		$widgets = get_option( 'sidebars_widgets', array() );
+
+		$widget_base = _get_widget_id_base( $request['id'] );
+		$widget_id = (int) substr( $request['id'], strlen( $widget_base ) + 1 );
+
+		$instances = get_option( 'widget_' . $widget_base, array() );
+
+		$instance = $instances[ $widget_id ];
+
+		$widget = new Milestone_Widget();
```

Comment:
Is the idea for this class meant to be expanded?  How will this look when/if we're using this to request data for other widgets?

Reply:
Yes, I have fixed that in the next commit - this was totally not ready.